wp/lib/bap_wp/src/run_parameters.ml

```diff
@@ -153,6 +162,27 @@ let validate (f : t) (files : string list) : (unit, error) result =
   validate_show f.show >>= fun () ->
   Ok ()
 
+(* Parses the loop invariant and address from the user into the format accepted
+   by the environment. *)
+let parse_loop_invariant (invariants : string) (sub : Sub.t)
+  : string Tid.Map.t =
+  if String.is_empty invariants then
+    Tid.Map.empty
+  else
+    Term.enum blk_t sub
+    |> Seq.fold ~init:Tid.Map.empty ~f:(fun map blk ->
+        match Term.get_attr blk address with
+        | None -> map
+        | Some address ->
+          let invs = invariant_list_of_sexp (Sexp.of_string invariants) in
+          List.fold invs ~init:map ~f:(fun m inv ->
+              let addr = Addr.of_string inv.address in
+              if Addr.equal address addr then
```

Comment:
Should we set a boolean here and print a warning if the address is *not* found?

Reply:
Sounds like a good idea.